pkg/api/2018-09-30-preview/api/types_test.go

```diff
@@ -1,12 +1,16 @@
-package api
+package api_test
```

Comment:
@jim-minter ran into that issue we discussed at the beginning of my time on the team....the cycles which can arise in testing and how `_test` packages help. Please feel free to suggest something different.

Reply:
call it `package api` and remove the import `. "github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"` below, then it should be fine.

Reply:
if I do that, I get the following error
*external*
```
# github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api
import cycle not allowed in test
package github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api (test)
	imports github.com/openshift/openshift-azure/test/util/populate
	imports github.com/openshift/openshift-azure/pkg/api
	imports github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api

FAIL	github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api [setup failed]
```

*internal*
```
/usr/local/go/bin/go test -c -i -o /tmp/___types_test_go__1_ github.com/openshift/openshift-azure/pkg/api #gosetup
# github.com/openshift/openshift-azure/pkg/api
import cycle not allowed in test
package github.com/openshift/openshift-azure/pkg/api (test)
	imports github.com/openshift/openshift-azure/test/util/populate
	imports github.com/openshift/openshift-azure/pkg/api

FAIL	github.com/openshift/openshift-azure/pkg/api [setup failed]
```